js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx

```diff
@@ -129,29 +257,30 @@ export function useIndexedDBCachedQuery<TQuery, TVariables extends OperationVari
       setError(error);
       setLoading(false);
     },
-    // exclude variables, instead JSON stringify it to avoid changing this reference if the caller hasn't memoized it
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [getData, client, key, query, JSON.stringify(variables), version, dataRef],
+    [dataRef, cacheRef],
   );
 
   React.useEffect(() => {
     if (skip) {
       return;
     }
-    getCachedData<TQuery>({key, version}).then((data) => {
+
+    cacheRef.getCachedData().then((data) => {
       if (data && !dataRef.current) {
         setData(data);
         setLoading(false);
       }
     });
-  }, [key, version, getCachedData, dataRef, skip]);
+  }, [dataRef, skip, cacheRef]);
 
   React.useEffect(() => {
     if (skip) {
       return;
     }
+    cacheRef.updateVariables(variables || ({} as TVariables));
+    cacheRef.updateVersion(version);
     fetch(true);
-  }, [fetch, skip]);
+  }, [fetch, skip, variables, version, cacheRef]);
```

Comment:
This is the only difference from https://github.com/dagster-io/dagster/pull/28629